src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Test-Json command.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "Json", HelpUri = "")]
+    public class TestJsonCommand : PSCmdlet
+    {
+        /// <summary>
+        /// An Json to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public String Json { get; set; }
+
+        /// <summary>
+        /// A schema to validate the Json against.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        public String Schema { get; set; }
+
+        private JSchema _jschema = null;
+
+        /// <summary>
+        /// Prepare an Json schema.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (Schema != null)
+            {
+                try
+                {
+                    _jschema = JSchema.Parse(Schema);
+                }
+                catch (Exception exc)
+                {
+                    Exception exception = new Exception(TestJsonCmdletStrings.InvalidJsonSchema, exc);
+                    ThrowTerminatingError(new ErrorRecord(exception, "InvalidJsonSchema", ErrorCategory.InvalidData, null));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validate an Json.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            JObject parsedJson = null;
+            bool result = true;
+
+            try
+            {
+                parsedJson = JObject.Parse(Json);
+
+                if (_jschema != null)
```

Comment:
I don't think this null check is necessary since we throw terminating error for invalid schema.

Reply:
Above we check whether Schema present - it is optional parameter. If Schema is absent here _jschema is null and we skip the test aganist a json schema.

Reply:
Schema should be a required "mandatory" property since the Json check cannot be run without it.  Make Schema mandatory and then the two explicit null checks are not necessary.

Reply:
Please disregard.